src/main/java/edu/umich/tl/analytics/CanvasCourseReportServlet.java

```diff
@@ -46,9 +51,12 @@
 	private static final String ACTION_GET_COURSES = "getCoursesPublished";
 	private static final String ACTION_GET_TERMS = "getEnrollmentTerms";
 	private static final String EMAIL_SUFFIX="@umich.edu";
+	private static final String ACTION_POLLING = "polling";
+	private static final String WORKING = "working";
 	private String canvasToken=CanvasCourseReportFilter.canvasToken;
 	private String canvasURL=CanvasCourseReportFilter.canvasURL;
 	ResourceBundle props = ResourceBundle.getBundle("coursereport");
+	ConcurrentHashMap<Thread,CourseReportTask> courseReportReqThreads=new ConcurrentHashMap<Thread,CourseReportTask>();
```

Comment:
The application will go on the tlsupport/tlsupportqa servers. As far i know their is only one server(not running in clustered environment) cutlass(qa)/matador(prod) running behind load balancer. This is a Tool for BSA and will be of low usage so storing in DB will be too much in compare to low usage. I like the session idea. can i have more opinions on this?